### PR TITLE
chore: added workspace files

### PR DESCRIPTION
**Pull request by AbdelrahmanEldabsha**
Migrations for workspace files
and Apis for (Create workspace, GeWorkspaces)

### CODE REVIEW

File: src/api/workspace/workspaceModel.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+// 	password: z.string(),
+// });
+// export type CreateUser = z.infer<typeof CreateUserSchema>;
+
+// export type CreateUserDto = Omit<User, 'id' | 'createdAt' | 'updatedAt'>;
+
+// // Input Validation for 'GET users/:id' endpoint
+// export const GetUserSchema = z.object({
+// 	params: z.object({ id: commonValidations.id }),
+// });
```
Comment (7adidaz) [inline src/api/workspace/workspaceModel.ts:70]: remove

File: src/api/workspace/workspaceRouter.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		tags: ['Workspace'],
+		security: [{ [bearerAuth.name]: [] }],
+		request: { params: GetWorkspaceSchema.shape.params },
+		responses: createApiResponse(WorkspaceSchema, 'Success'),
+	});
+
+	router.get('/:id', validateRequest(GetWorkspaceSchema), WorkspaceController.getWorkspaceById);
+
+	return router;
+})();
```
Comment (7adidaz) [inline src/api/workspace/workspaceRouter.ts:64]: Consider adding an **authentication** to the middleware array and create an **authorization** middleware, I tried creating a `workspace` with a different `ownerId` than mine and it worked, the same goes for creating a `channel`. 

I also expect that users outside the `workspace` or `channel` can't retrieve data about the channel, you can use the `members` table to check for access and create a new table representing the `many to many` relationship between users and workspaces, which you can use later as a reference for the authorization step.


```
{
  "success": false,
  "message": "Error creating Workspace: insert into `workspaces` (`avatarUrl`, `description`, `name`, `ownerId`) values ('string', 'string', 'string', 1) - Cannot add or update a child row: a foreign key constraint fails (`wavely`.`workspaces`, CONSTRAINT `workspaces_ownerid_foreign` FOREIGN KEY (`ownerId`) REFERENCES `users` (`id`))",
  "responseObject": null,
  "statusCode": 500
}
```

**Review by 7adidaz — CHANGES_REQUESTED** [review]

File: src/api/channels/channelController.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+		handleServiceResponse(serviceResponse, res);
+	},
+	getChannelById: async (req: Request, res: Response) => {
+		const id = parseInt(req.params.id);
+		const serviceResponse = await channelService.findChannelById(id);
+		handleServiceResponse(serviceResponse, res);
+	},
+	createChannel: async (req: Request, res: Response) => {
+		const { name, description, creatorId, type, workspaceId } = req.body;
+
```
[the remark this answers is not in the document]
Comment (AbdelrahmanEldabsha) [reply src/api/channels/channelController.ts:20]: Its creatorId in db schema, its not like user schema

File: src/api/channels/channelRouter.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	channelRegistery.registerPath({
+		method: 'get',
+		path: '/channels/getWorkspaceChannel/{id}',
+		tags: ['Channel'],
+		security: [{ [bearerAuth.name]: [] }],
+		request: { params: GetChannelSchema.shape.params },
+		responses: createApiResponse(ChannelSchema, 'Success'),
+	});
+
+	router.get('/getWorkspaceChannel/:id', AuthController.authenticate, ChannelController.getChannels);
```
Comment (7adidaz) [inline src/api/channels/channelRouter.ts:56]: Better is to use `/workspace/:id/channels` looks more restful which will be used and routed by the workspace controller, and of course cancel the `ChannelController.getChannels` controller and use the internal service response with `findAllWorkspaceChannels` 

https://opensource.zalando.com/restful-api-guidelines/#138

**Review by 7adidaz — CHANGES_REQUESTED** [review]

**Pull request merged**